src/cookieSync.js

```diff
@@ -14,33 +14,49 @@ const MAX_SYNC_COUNT = sanitizeSyncCount(parseInt(parseQueryParam('max_sync_coun
  */
 const isValidUrl =  new RegExp(/^(?:(?:(?:https?|ftp):)?\/\/)(?:\S+(?::\S*)?@)?(?:(?!(?:10|127)(?:\.\d{1,3}){3})(?!(?:169\.254|192\.168)(?:\.\d{1,3}){2})(?!172\.(?:1[6-9]|2\d|3[0-1])(?:\.\d{1,3}){2})(?:[1-9]\d?|1\d\d|2[01]\d|22[0-3])(?:\.(?:1?\d{1,2}|2[0-4]\d|25[0-5])){2}(?:\.(?:[1-9]\d?|1\d\d|2[0-4]\d|25[0-4]))|(?:(?:[a-z0-9\u00a1-\uffff][a-z0-9\u00a1-\uffff_-]{0,62})?[a-z0-9\u00a1-\uffff]\.)+(?:[a-z\u00a1-\uffff]{2,}\.?))(?::\d{2,5})?(?:[/?#]\S*)?$/i);
 
-function doBidderSync(type, url, bidder) {
+function doBidderSync(type, url, bidder, done) {
   if (!url || !isValidUrl.test(url)) {
     console.log(`No valid sync url for bidder "${bidder}": ${url}`);
+    done();
   } else if (type === 'image' || type === 'redirect') {
     console.log(`Invoking image pixel user sync for bidder: "${bidder}"`);
-    triggerPixel(url);
+    triggerPixel(url, done);
   } else if (type == 'iframe') {
+    console.log(`Skipping iframe pixel user sync for bidder: "${bidder}". This isn't implemented yet.`);
     // TODO test iframe solution
+    done();
   } else {
     console.log(`User sync type "${type}" not supported for bidder: "${bidder}"`);
+    done();
   }
 }
 
-function triggerPixel(url) {
+function triggerPixel(url, done) {
   const img = new Image();
+  img.addEventListener('load', done);
+  img.addEventListener('error', done);
   img.src = url;
 }
 
+function doAllSyncs(bidders) {
+  if (bidders.length === 0) {
+    return;
+  }
+
+  const thisSync = bidders[bidders.length - 1];
+  bidders = bidders.slice(0, bidders.length - 1);
```

Comment:
As a small question - wouldn't you be able to use `const thisSync = bidders.pop()` instead of using both of these statements?

Reply:
yes! I didn't know that was a function in JS. Definitely better that way, though.